sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredential.java

```diff
@@ -38,27 +39,26 @@
     public Mono<AccessToken> getToken(TokenRequestContext request) {
         List<CredentialUnavailableException> exceptions = new ArrayList<>(4);
         return Flux.fromIterable(credentials)
-                   .flatMap(p -> p.getToken(request).onErrorResume(CredentialUnavailableException.class, t -> {
-                       exceptions.add(t);
+                   .flatMap(p -> p.getToken(request).onErrorResume(Exception.class, t -> {
+                       if (!t.getClass().getSimpleName().equals("CredentialUnavailableException")) {
+                           return Mono.error(new CredentialUnavailableException(
```

Comment:
Why are we wrapping a non Credential Unavailable Exception into a Credential Unavailable Exception ?
It should be thrown as it is, to indicate a genuine exception even though credential is available to be used.
The distinction will help to debug customer issues too in future.